### PR TITLE
Add Clean Command

### LINKED ISSUE #96: Add parser support for clean command

### LINKED ISSUE #97: As an organized dorm manager, I can delete residents who have already graduated from my address book so that my address book remains organized

### LINKED ISSUE #103: Add CleanCommand class

### LINKED ISSUE #104: Add Predicate for graduation date

### PR DESCRIPTION
**Pull request by EdaYeo**
* Add parser support for clean command
* Add clean command class
* Add GradYear predicate

Fixes #97, fixes #96, fixes #103, fixes #104

### CODE REVIEW

**Review by lancehaha — COMMENTED** [review]
it would be better to add method comment to pass the gradlew test

File: src/test/java/seedu/address/logic/commands/CleanCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+/**
+ * Contains integration tests (interaction with the Model) and unit tests for
+ * {@code CleanCommand}.
+ */
+public class CleanCommandTest {
+    private Model model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+
+    @Test
+    public void execute_clean_success() {
```
Comment (charlesliyifeng) [inline src/test/java/seedu/address/logic/commands/CleanCommandTest.java:25]: You are not supplying a predicate in this test case, so expectedModel always has an empty address book?

I am assuming that you will add the predicate after clean command is implemented.

**Review by charlesliyifeng — COMMENTED** [review]
I have a question on the implementation of clean command.
But other than that, everything else looks ok to me.

File: src/main/java/seedu/address/logic/parser/AddressBookParser.java [hunk 2]
```diff
@@ -17,7 +18,6 @@
 import seedu.address.logic.commands.FindCommand;
 import seedu.address.logic.commands.HelpCommand;
 import seedu.address.logic.commands.ListCommand;
-import seedu.address.logic.commands.UndoCommand;
```
Comment (charlesliyifeng) [inline src/main/java/seedu/address/logic/parser/AddressBookParser.java:20]: Why remove undo command?

File: src/test/java/seedu/address/logic/commands/CleanCommandTest.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        CommandResult expectedCommandResult = new CommandResult(MESSAGE_CLEAN_SUCCESS);
+        ModelManager expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+        List<Person> lastShownList = expectedModel.getFilteredPersonList();
+        int listSize = lastShownList.size();
+        for (int i = listSize - 1; i >= 0; i--) {
+            Person p = lastShownList.get(i);
+            Optional<GradYear> graduationYear = p.getGradYear();
+            if (graduationYear.isPresent()) {
+                String theYear = graduationYear.get().toString();
+                System.out.println(theYear);
```
Comment (charlesliyifeng) [inline src/test/java/seedu/address/logic/commands/CleanCommandTest.java:44]: Perhaps the println should be removed?

**Review by charlesliyifeng — APPROVED** [review]
LGTM

File: src/test/java/seedu/address/logic/commands/CleanCommandTest.java (hunk 1 above)
Comment (lancehaha) [reply src/test/java/seedu/address/logic/commands/CleanCommandTest.java:34]: Any predicate needed? Clean Command creates its own predicate, and expected model has deleted the graduated people.

**Pull request merged**